pennylane/templates/embeddings/basis.py

```diff
@@ -88,10 +96,12 @@ def __init__(self, features, wires, do_queue=True, id=None):
                 f"Features must be of length {len(wires)}; got length {n_features} (features={features})."
             )
 
-        features = list(qml.math.toarray(features))
-
-        if not set(features).issubset({0, 1}):
-            raise ValueError(f"Basis state must only consist of 0s and 1s; got {features}")
+        # don't perform the validation if the features is a JAX tracer or a TensorFlow
+        # non-eager tensor
+        if not qml.math.is_abstract(features):
```

Comment:
It looks like we might evaluate `not qml.math.is_abstract(features)` twice in certain cases in this method. How about assigning it to an object earlier on and using that object at the two occurrences?

Reply:
Fair, updated.